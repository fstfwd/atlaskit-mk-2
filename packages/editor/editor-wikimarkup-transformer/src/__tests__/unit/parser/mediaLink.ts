import WikiMarkupTransformer from '../../../index';

describe('JIRA wiki markup - mediaLink', () => {
  const testCases: Array<[string, string]> = [
    ['should create a mediaLink in mediaGroup node', '[^test.txt]'],
    [
      'should not include mediaLink inside a list',
      '* some random list [^not-in-list.txt]',
    ],
  ];

  for (const [testCaseDescription, markup] of testCases) {
    it(testCaseDescription, () => {
      const transformer = new WikiMarkupTransformer();
      expect(transformer.parse(markup)).toMatchSnapshot();
    });
  }
});
