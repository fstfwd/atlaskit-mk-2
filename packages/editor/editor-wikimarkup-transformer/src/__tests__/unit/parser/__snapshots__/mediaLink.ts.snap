// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JIRA wiki markup - mediaLink should create a mediaLink in mediaGroup node 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "attrs": Object {
            "__displayType": null,
            "__fileMimeType": null,
            "__fileName": null,
            "__fileSize": null,
            "__key": null,
            "collection": "",
            "height": null,
            "id": "test.txt",
            "occurrenceKey": null,
            "type": "file",
            "url": null,
            "width": null,
          },
          "type": "media",
        },
      ],
      "type": "mediaGroup",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - mediaLink should not include mediaLink inside a list 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "* some random list ",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "content": Array [
        Object {
          "attrs": Object {
            "__displayType": null,
            "__fileMimeType": null,
            "__fileName": null,
            "__fileSize": null,
            "__key": null,
            "collection": "",
            "height": null,
            "id": "not-in-list.txt",
            "occurrenceKey": null,
            "type": "file",
            "url": null,
            "width": null,
          },
          "type": "media",
        },
      ],
      "type": "mediaGroup",
    },
  ],
  "type": "doc",
}
`;
